apps/core/lib/core/snitch/variant.ex

```diff
@@ -8,21 +8,20 @@ defmodule Core.Snitch.Variant do
   @type t :: %__MODULE__{}
   schema "snitch_variants" do
     field(:sku, :string, default: "")
-    field(:weight, :integer, precision: 8, scale: 2, default: 0.0)
-    field(:height, :integer, precision: 8, scale: 2)
-    field(:width, :integer, precision: 8, scale: 2)
-    field(:depth, :integer, precision: 8, scale: 2)
+    field(:weight, :decimal, precision: 8, scale: 2, default: 0)
+    field(:height, :decimal, precision: 8, scale: 2)
+    field(:width, :decimal, precision: 8, scale: 2)
+    field(:depth, :decimal, precision: 8, scale: 2)
     field(:is_master, :boolean, default: false)
-    field(:cost_price, :integer, precision: 10, scale: 2)
+    field(:cost_price, Money.Ecto.Composite.Type)
```

Comment:
Since we have a `snitch_prices` table, can we skip this?

Reply:
According to this issue on https://github.com/spree/spree/issues/4734, `cost_price` is the cost price. Selling prices will be in `snitch_prices`. The issue also exposes a limitation in how the stock is managed and is worthy of discussion on its own.